File: 04-Secuencial/05-Huella/02-Go/modulo/util-old.go
/*
   Módulo de utilidades (funciones de apoyo)
   Autor: Julián Esteban Gutiérrez Posada
   Fecha: Marzo 2025
   Licencia: GNU GPL v3
*/

package modulo

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// MostrarMensaje mostrar un mensaje (cadena) en la salida estandar.
func MostrarMensaje(mensaje string) {
	fmt.Print(mensaje)
}

// IngresarTexto muestra una pregunta y devuelve el texto ingresado por el usuario.
func IngresarTexto(pregunta string) string {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print(pregunta)
	texto, _ := reader.ReadString('\n')
	return strings.TrimSpace(texto)
}

// IngresarEntero muestra una pregunta y devuelve el entero ingresado por el usuario o cero si es un valor inválido.
func IngresarEntero(pregunta string) int {
	entero, _ := strconv.Atoi(IngresarTexto(pregunta))
	return entero
}

// IngresarReal muestra una pregunta y devuelve el real ingresado por el usuario o cero si es un valor inválido.
func IngresarReal(pregunta string) float64 {
	real, _ := strconv.ParseFloat(IngresarTexto(pregunta), 64)
	return real
}

File: 04-Secuencial/05-Huella/04-Javascript/modulo/util-old.js
/*
   Módulo de utilidades (funciones de apoyo)
   Autor: Julián Esteban Gutiérrez Posada
   Fecha: Marzo 2025
   Licencia: GNU GPL v3
*/

/**
 * Muestra un mensaje (cadena) en la salida estandar.
 * 
 * @param {string} mensaje - Mensaje que se desea mostrar.
 */
function mostrarMensaje(mensaje) {
  document.getElementById("salida").value = mensaje;
}

/**
 * Devuelve el texto ingresado por el usuario como respuesta a una pregunta
 * 
 * @param {string} componente Nombre del componente donde el usuario 
 *                            ingresa la respuesta
 * @return {string} Texto ingresado por el usuario
 */
function ingresarTexto(componente) {
  return document.getElementById(componente).value;
}

/**
 * Devuelve el entero ingresado por el usuario como respuesta a una 
 * pregunta.
 * 
 * @param {string} componente Nombre del componente donde el usuario 
 *                            ingresa la respuesta
 * @return {int} Valor ingresado por el usuario o cero si es un valor 
 *               inválido.
 */
function ingresarEntero(componente) {
  return parseInt(ingresarTexto(componente));
}

/**
 * Devuelve el real ingresado por el usuario como respuesta a una pregunta.
 * 
 * @param {string} componente Nombre del componente donde el usuario 
 *                            ingresa la respuesta
 * @return {float} Valor ingresado por el usuario o cero si es un valor
 *                 inválido.
 */
function ingresarReal(componente) {
  return parseFloat(ingresarTexto(componente));
}